util/merkle-mountain-range/src/merge.rs

```diff
@@ -0,0 +1,6 @@
+use crate::Result;
+
+pub trait Merge {
+    type Item;
+    fn merge(left: &Self::Item, right: &Self::Item) -> Result<Self::Item>;
```

Comment:
I can't see the reason for using `Result` here, `GetRootOnEmpty` and `InconsistentStore` are not related to `merge` function.

Reply:
I want to maintain some basic abstract on `Merge` trait, some error is possible to happen on merge in logically like `total_difficulty` overflow, so the MMR can return an error from `Merge`.